mantidproject/mantid
# Regression and additional fixes for NRCalculateSlitResolution

This PR fixes a regression in the old reflectometry GUI following changes made for #18442. It also adds some additional minor fixes that had been missed in the original changes for #18442.

In more detail, the changes are:
- Fixed a regression where the old reflectometry GUI was throwing because it was still looking for `CalculateResolution`, which has been renamed to `NRCalculateSlitResolution`.
- Fixed a problem where the old GUI was trying to get theta from `NRCalculateSlitResolution` but it is no longer returned. It now gets theta from the logs instead (from the same log value that `CalcualteResolution` was using previously, so the behaviour is the same).
- Fixed a bug in both the old and new GUIs where theta was being passed to `NRCalculateSlitResolution` instead of twoTheta. This is in line with other fixes in #18442 where theta was being misused as twoTheta.
- Fixed some comments that were still referring to `CalculateResolution`.
- Fixed some confusing variable naming in the old GUI where theta was being called two_theta.

Fixes #20529 

**To test:**
* Run the script below.
* Check results with spreadsheet in #18442:
  * Note the values that are printed for `s1`, `s2` and `theta` for any of the runs
  * Put these values into the boxes with the same names in the spreadsheet.
  * Check the value of `dQ/Q` in the spreadsheet. This should be the same as the result printed by the script as a percentage (to 5 or 6dp).
* Old GUI:
  * Open the old reflectometry GUI (Interfaces -> Reflectometry -> ISIS Reflectometry Old).
  * On the first row, enter `13460` in the first `Run(s)` box and `13461+13462` in the second `Run(s)` box, then hit `Process`.
  * Check that the `Angle1` box gets populated with `0.700020` and `Angle2` with `2.299999` (the same values printed by the script for the first and last runs) and that the `dQ/Q` field gets set to `0.0340274472002` (the same resolution as the first run printed by the script)
* New GUI
  * Open the new reflectometry GUI (Interfaces -> Reflectometry -> ISIS Reflectometry).
  * Set the instrument to `INTER`, enter Investigation ID `1120015` and hit `Search`.
  * Select runs `13460`, `13461` and `13462` in the list and hit `Transfer`.
  * Set the angle for the first row to `0.700020` and for the second to `2.299999`, then process the group.
  * Check that the `dQ/Q` values that get populated in the two rows are the same as those printed by the script for the first and last runs respectively.

```
print "Put s1, s2 and theta into the spreadsheet in issue #18442 and you"
print "should get the same result for dQ/Q."
print "https://github.com/mantidproject/mantid/issues/18442"

def test(name):
    print
    print name
    ws = Load(name)
    theta = ws.getRun().getLogData("Theta").value[-1]
    print "theta =  %.12f" % (theta)
    print "s1 = %.6f" % (ws.getRun().getLogData("S1_VG").value[-1])
    print "s2 = %.6f" %(ws.getRun().getLogData("S2_VG").value[-1])
    
    result = NRCalculateSlitResolution(Workspace = ws)
    print "dQ/Q = %.13f" % (result)
    print "     = %.6f%%" % (result*100)
    
    check = NRCalculateSlitResolution(Workspace = ws, TwoTheta=2*theta)
    if check != result:
        print "ERROR, different result when passing TwoTheta property"
    check = NRCalculateSlitResolution(Workspace = ws, ThetaLogName="Theta")
    if check != result:
        print "ERROR, different result when passing ThetaLogName property"

test('INTER00013460')    
test('INTER00013461')
test('INTER00013462')
```

**Release Notes** 
*Added to the release notes*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
